tools/mill/cmd/googlecloud.go

```diff
@@ -256,6 +278,120 @@ func removeSubscription(id string) error {
 	return sub.Delete(ctx)
 }
 
+func listSubscriptions(topic string, adapter watermill.LoggerAdapter, verbose bool) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	client, err := pubsub.NewClient(ctx, projectID())
+	if err != nil {
+		return errors.Wrap(err, "could not create pubsub client")
+	}
+
+	if topic != "" {
+		topic := client.Topic(topic)
+		return listSubscriptionsForTopic(ctx, client, topic, logger, verbose)
+	}
+
+	it := client.Topics(ctx)
+	noTopics := true
+	for {
+		topic, err := it.Next()
+		if err == iterator.Done {
+			if noTopics {
+				logger.Info("No topics in project", nil)
+			}
+			return nil
+		}
+		if err != nil {
+			return errors.Wrap(err, "could not retrieve next subscription")
+		}
+
+		noTopics = false
+		err = listSubscriptionsForTopic(ctx, client, topic, logger, verbose)
+		if err != nil {
+			return errors.Wrap(err, "error listing subscriptions for topic")
+		}
+	}
+
+	return nil
+}
+
+func listSubscriptionsForTopic(ctx context.Context, client *pubsub.Client, topic *pubsub.Topic, logger watermill.LoggerAdapter, verbose bool) error {
+	noSubs := true
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not check if topic exists")
+	}
+	if !exists {
+		logger.Info("Topic does not exist", watermill.LogFields{"topic": topic.String()})
+		return nil
+	}
+
+	it := topic.Subscriptions(ctx)
+	for {
+		sub, err := it.Next()
+		if err == iterator.Done {
+			if noSubs {
+				logger.Info("No subscriptions for the topic", watermill.LogFields{"topic": topic.String()})
+			}
+			return nil
+		}
+		if err != nil {
+			return errors.Wrap(err, "could not retrieve next subscription")
+		}
+
+		if noSubs {
+			noSubs = false
+			fmt.Printf("Topic %s:\n", topic.String())
+		}
+		name := sub.String()
+		config, err := sub.Config(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "could not retrieve subscription config for subscription '%s'", name)
+		}
+
+		err = printSubscriptionInfo(name, config)
+		if err != nil {
+			return errors.Wrapf(err, "error printing subscription '%s'", name)
+		}
+	}
+}
+
+func printSubscriptionInfo(name string, config pubsub.SubscriptionConfig) error {
+	b, err := yaml.Marshal(subscriptionConfig{
```

Comment:
why YAML actually? maybe some ASCII table would be a bit better? 🤔

Reply:
I chose YAML because it's more readable than JSON.
asciitable should be OK as well.

It will be adding a new dependency, though. We already depend on yaml for some reason.